cortex/secondary/call_duration.py

```diff
@@ -27,15 +29,28 @@ def call_duration(incoming=True, **kwargs):
             timestamp (int): The beginning of the window (same as kwargs['start']).
             value (float): The time spent in a call.
     """
-    incoming_dict = {True: 1, False: 2}
-    label = incoming_dict[incoming]
-    log.info('Loading raw calls data...')
-    _calls = calls(**kwargs)['data']
-    log.info('Computing call duration...')
-    _call_duration = np.sum([call['call_duration'] for call in _calls
-                             if call['call_type'] == label])
+    log.info('Computing call duration ...')
+
+    _calls = telephony(**kwargs)['data']
+
+    call_duration = np.sum(call['duration'] for call in _calls)
+
+    if incoming == "all":
```

Comment:
Since the only possible `type` should be 'incoming' or 'outgoing' (https://docs.lamp.digital/data_science/data_types/sensor_types/#telephony) I think you may need only one of line 36 or the block at line 38, as either of them should cover the `all` case.